pandas/tests/internals/test_external_block.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=W0102
+
+import numpy as np
+
+import pandas as pd
+from pandas.core.internals import Block, BlockManager
+
+
+class CustomBlock(Block):
+
+    def formatting_values(self):
+        return np.array(["Val: {}".format(i) for i in self.values])
+
+
+def test_custom_repr():
+    values = np.arange(3, dtype='int64')
+
+    # series
+    block = CustomBlock(values, placement=slice(0, 3))
```

Comment:
yes, that's fine. But slightly change my comment: it has also proven to be difficult to add a block to a BlockManager with preserving the block type.

Once we have a BlockManager, it's indeed simply passing it to DataFrame(..) to get a df. That's what we do to create dataframes, I should probably take the same approach to create SingleBlockManager for the series case instead of using that fastpath.